lib_eio/mock/net.ml

```diff
@@ -71,17 +71,51 @@ let on_getaddrinfo (t:t) actions = Handler.seq t#on_getaddrinfo actions
 
 let on_getnameinfo (t:t) actions = Handler.seq t#on_getnameinfo actions
 
+type stream_socket = <
+  Flow.t;
+  Eio.Net.stream_socket; 
+  on_setsockopt: unit Handler.t;
+>
+
 type listening_socket = <
   Eio.Net.listening_socket;
-  on_accept : (Flow.t * Eio.Net.Sockaddr.stream) Handler.t;
+  on_accept : (stream_socket * Eio.Net.Sockaddr.stream) Handler.t;
+  on_setsockopt: unit Handler.t;
 >
 
+let setsockopt (type a) label handler (opt: a Eio.Net.sockopt) (x : a) : unit =
+  let traceln (pp : a Fmt.t) (x: a) s = 
+    traceln "%s: setsockopt %s %a" label s pp x 
+  in
+  match opt with
+  | IPV6_ONLY     -> traceln Fmt.bool x "IPV6_ONLY"
+  | SO_ACCEPTCONN -> traceln Fmt.bool x "SO_ACCEPTCONN"
+  | SO_BROADCAST  -> traceln Fmt.bool x "SO_BROADCAST"
+  | SO_DEBUG      -> traceln Fmt.bool x "SO_DEBUG"
+  | SO_DONTROUTE  -> traceln Fmt.bool x "SO_DONTROUTE"
+  | SO_KEEPALIVE  -> traceln Fmt.bool x "SO_KEEPALIVE"
+  | SO_LINGER     -> traceln Fmt.(option int) x "SO_LINGER"
+  | SO_OOBINLINE  -> traceln Fmt.bool x "SO_OOBINLINE"
+  | SO_RCVBUF     -> traceln Fmt.int x "SO_RCVBUF"
+  | SO_RCVLOWAT   -> traceln Fmt.int x "SO_RCVLOWAT"
+  | SO_REUSEADDR  -> traceln Fmt.bool x "SO_REUSEADDR"
+  | SO_REUSEPORT  -> traceln Fmt.bool x "SO_REUSEPORT"
+  | SO_SNDBUF     -> traceln Fmt.int x "SO_SNDBUF"
+  | SO_SNDLOWAT   -> traceln Fmt.int x "SO_SNDLOWAT"
+  | SO_RCVTIMEO   -> traceln Fmt.float x "SO_RCVTIMEO"
+  | SO_SNDTIMEO   -> traceln Fmt.float x "SO_SNDTIMEO"
+  | SO_TYPE       -> traceln Fmt.int x "SO_TYPE"
+  | TCP_NODELAY   -> traceln Fmt.bool x "TCP_NODELAY"
+  ;
```

Comment:
You can't use `;` to end a match block (try ocp-indent on the code and it will show the problem).

Reply:
good catch. Fixed now.